include/exadg/postprocessor/pointwise_output_generator_base.h

```diff
@@ -0,0 +1,198 @@
+             dealii::Mapping<dim> const &         mapping_in,
+             PointwiseOutputDataBase<dim> const & pointwise_output_data_in);
+
+  void
+  evaluate(VectorType const & solution, double const & time, int const & time_step_number);
+
+  void
+  add_scalar_quantity_dataset(std::string const & name);
+
+  void
+  add_vector_quantity_datasets(std::string const & name);
+
+  void
+  write_scalar_quantity(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
+
+  void
+  write_vectorial_quantity(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
+
+  void
+  write_scalar_quantity(std::string const &         name,
+                        std::vector<Number> const & values,
+                        unsigned int const          first_component = 0)
+  {
+    AssertThrow(first_component == 0,
+                dealii::ExcMessage("This function can only be called with first_component==0."));
+
+    hyperslab_data = std::move(dealii::Vector<Number>(values.begin(), values.end()));
+    write_hyperslab(name, hyperslab_data);
+  }
+
+  template<int n_components>
+  void
+  write_scalar_quantity(std::string const &                                          name,
+                        std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                        unsigned int const first_component = 0)
+  {
+    this->extract_component(hyperslab_data, values, first_component);
+    write_hyperslab(name, hyperslab_data);
+  }
+
+  template<int n_components>
+  void
+  write_vectorial_quantity(std::string const &                                          name,
+                           std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                           unsigned int const first_component = 0)
+  {
+    for(unsigned int d = 0; d < dim; ++d)
+    {
+      this->extract_component(hyperslab_data, values, d + first_component);
+      write_hyperslab(name + std::to_string(d), hyperslab_data);
+    }
+  }
+
+  template<int n_components>
+  [[nodiscard]] decltype(auto) // TODO: be explicit about return
+  compute_point_values(VectorType const & solution) const
+  {
+    return dealii::VectorTools::point_values<n_components>(*remote_evaluator,
+                                                           *dof_handler,
+                                                           solution);
+  }
+
+private:
+  virtual void
+  do_evaluate(VectorType const & solution) = 0;
+
+  void
+  setup_remote_evaluator();
+
+  void
+  reinit_remote_evaluator();
+
+  void
+  create_hdf5_file();
+
+  void
+  add_evaluation_points_dataset(dealii::HDF5::Group & group, std::string const & name);
+
+  void
+  fill_evaluation_points(std::string const & name);
+
+  void
+  add_time_dataset(dealii::HDF5::Group & group, std::string const & name);
+
+  void
+  write_time(double time);
+
+  void
+  write_hyperslab(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
```

Comment:
The naming `hyperslab` is very prominent in this MR. Does this come from HDF5? What is the meaning? I wonder whether we could find a more expressive name ...

Reply:
It comes from HDF5. A hyperslab is basically the place where to write data. So hyperslab data would be the data that has to be written to the hyperslab. In this case, it would be the values of a component at all points at a certain time. So ```componentwise_result``` might be a name we could use if you don't want to have the HDF5 specific names here.

Reply:
Yes, I would prefer such a name.

Reply:
resolved in [bddce88](https://github.com/exadg/exadg/pull/255/commits/bddce8854304c2194e17aaa84d9d06e5fef4d23f)